From fb513173d56809ef76f7a9a260c104b0b10f45cd Mon Sep 17 00:00:00 2001
From: Katsuya Matsubara <matsu@igel.co.jp>
Date: Mon, 17 Dec 2012 15:32:23 +0900
Subject: [PATCH 31/39] base_videodec: add a hook function for state
 transition

This function allows to describe extra preparation or
finalization especially for video decoder since it could be
invoked whenever the state would be changed.
---
 omx/gstomx_base_videodec.c |   14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/omx/gstomx_base_videodec.c b/omx/gstomx_base_videodec.c
index 6742927..1ff5e95 100644
--- a/omx/gstomx_base_videodec.c
+++ b/omx/gstomx_base_videodec.c
@@ -31,9 +31,23 @@ type_base_init (gpointer g_class)
 {
 }
 
+static GstStateChangeReturn
+change_state (GstElement * element, GstStateChange transition)
+{
+  GstStateChangeReturn ret;
+
+  ret = GST_ELEMENT_CLASS (parent_class)->change_state (element, transition);
+
+  return ret;
+}
+
 static void
 type_class_init (gpointer g_class, gpointer class_data)
 {
+  GstElementClass *gstelement_class;
+
+  gstelement_class = GST_ELEMENT_CLASS (g_class);
+  gstelement_class->change_state = change_state;
 }
 
 static gboolean
-- 
1.7.9.5

